File: apps/api/app/schemas/rbac.py
from typing import List, Optional
from pydantic import BaseModel


class PermissionSchema(BaseModel):
    id: str
    code: str
    name: str
    description: str
    group: str


class RoleResponse(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    is_system_default: bool
    organization_id: str
    permissions: List[PermissionSchema]
    created_at: str


class CreateRoleRequest(BaseModel):
    name: str
    description: Optional[str] = None
    permission_ids: List[str]

File: packages/types/src/rbac.ts
export enum SystemDefaultRole {
  OWNER = 'Owner',
  ADMINISTRATOR = 'Administrator',
  NETWORK_ENGINEER = 'Network Engineer',
  OPERATOR = 'Operator',
  READ_ONLY = 'Read Only',
}

export interface Permission {
  id: string;
  code: string;
  name: string;
  description: string;
  group: string;
}

export interface Role {
  id: string;
  name: string;
  description?: string;
  isSystemDefault: boolean;
  organizationId: string;
  permissions: Permission[];
  createdAt: string;
  updatedAt: string;
}

export interface CreateRolePayload {
  name: string;
  description?: string;
  permissionIds: string[];
}
